src/index.ts

```diff
@@ -50,22 +57,49 @@ export function useFlexSearch <T extends {id: string| number}> (
         index.value = providedIndex.value
       }
     },
-    { immediate: true }
+    {
+      immediate: true
+    }
   )
 
   return computed(() => {
     if (!query || !index.value || !store) return []
     if (query) {
       const rawResults = index.value.search(query.value, limit, searchOptions)
 
-      return rawResults.map(rawResult => {
-        if (isDocument.value) {
-          const [id] = (rawResult as SimpleDocumentSearchResultSetUnit).result
+      if (isDocument.value) {
+        const results: T[] = []
+        const resultIds = new Set<number | string>()
+
+        for (const rawResult of rawResults) {
+          if (isSimpleDocumentSearchResultSetUnit(rawResult)) {
+            for (const id of rawResult.result) {
+              if (!resultIds.has(id)) {
+                resultIds.add(id)
+                const match = store.value.find((item: T) => item.id === id)
 
-          return store.value.find(storeItem => storeItem.id === id)
+                if (match) {
+                  results.push(match)
+                }
+              }
+            }
+          }
+        }
+        return results
+      }
+      return rawResults.map(rawResult => {
+        if (!isSimpleDocumentSearchResultSetUnit(rawResult)) {
+          return store.value.find(storeItem => storeItem.id === rawResult)
         }
-        return store.value.find(storeItem => storeItem.id === rawResult)
       })
     }
   })
 }
+function isSimpleDocumentSearchResultSetUnit (
+  object: FlexSearch.Id | FlexSearch.SimpleDocumentSearchResultSetUnit
+): object is FlexSearch.SimpleDocumentSearchResultSetUnit {
+  return (
+    (object as FlexSearch.SimpleDocumentSearchResultSetUnit).result !==
```

Comment:
@LwveMike Is this ok in you opinion?

Reply:
Using 'as' in a type predicate defeats its purpose.